src/Aspire.Hosting/Dashboard/DashboardInnerLoopHook.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal sealed class DashboardInnerLoopHook(ILogger<DashboardInnerLoopHook> logger, DashboardServiceHost dashboardHost) : IDistributedApplicationLifecycleHook
+{
+    private readonly DashboardServiceHost _dashboardHost = dashboardHost;
+    private readonly ILogger<DashboardInnerLoopHook> _logger = logger;
+
+    public Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
+    {
+        if (appModel.Resources.SingleOrDefault(r => StringComparers.ResourceName.Equals(r.Name, KnownResourceNames.AspireDashboard)) is not { } dashboardResource)
+        {
+            _logger.LogDebug("Dashboard resource not found in app model. Skipping dashboard hook.");
+            return Task.CompletedTask;
+        }
+
+        var innerloopLaunchProfile = new LaunchProfileAnnotation("innerloop");
+        dashboardResource.Annotations.Add(innerloopLaunchProfile);
```

Comment:
Could you provide a little hint here to connect with the comment you wrote about the `innerloop` launch profile? It's not clear to me whether the `"innerloop"` literal here is intended to match that launch profile name, and if so then what's the significance of that?

For example, guessing a bit:

```suggestion
        // We only want to be called back for resources annotated with "innerloop".
        dashboardResource.Annotations.Add(new LaunchProfileAnnotation("innerloop"));
```

(Also inlined the var to match the next two calls.)

Reply:
No longer required.